src/qforte/abc/algorithm.py

```diff
@@ -258,6 +260,21 @@ def fill_pool(self):
         else:
             raise ValueError('Invalid operator pool type specified.')
 
+        # If possible, impose symmetry restriction to operator pool
```

Comment:
I have doubts about whether this code belongs in the `Algorithm` class, but I can deal with this later. A PR I have planned will clarify the matter.

Reply:
Ideally, this code should have been part of the C++ function `fill_pool`. The way the code works now is not the most efficient one: first the `SQOpPool` is filled in the C++ side of the code, without any symmetry restrictions, and then symmetry is imposed in Python. Placing this code in the C++ function `fill_pool` will also allow us to remove the `try` and `except` statements, since the line `elif isinstance(self._pool_type, qf.SQOpPool):` does not involve the C++ `fill_pool` function. To be able to do this, we would need to pass the information regarding orbital irreps from Python to C++. My current knowledge (or lack thereof) of both Python and C++ prevents me from doing this. If you feel that this is the best solution, I would be happy to discuss how to proceed.

Reply:
Let's save that for a future PR.